src/python/grpcio/grpc/_cython/_cygrpc/aio/iomgr/io_loop.pyx.pxi

```diff
@@ -0,0 +1,87 @@
+# Copyright 2019 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sys
+import threading
+
+_RESOLUTION_MS = 0.001
+cdef _IOLoop _io_loop = None
+
+cdef class _IOLoop:
+
+    def __init__(self):
+        global _io_loop
+
+        # Only one instantiation is exepected        
+        assert _io_loop is None
+
+        self._asyncio_loop = None
+        self._io_ev = threading.Event()
+        self._loop_started_cv = threading.Condition()
+        self._loop_started = False
+
+        self._thread = threading.Thread(target=self._run_forever, daemon=True)
+        self._thread.start()
+
+        # Some attributes are initialitzated when
+        # the thread is really started, we wait till
+        # the whole Asyncio loop is initialized and
+        # ready to be used.
+        with self._loop_started_cv:
```

Comment:
If this used a `threading.Event` instead, this could just be `self._loop_started_event.wait()`.

Reply:
Yes, I will migrate this for using `Event`. Thanks!